src/map/sortLayers.js

```diff
@@ -12,106 +12,79 @@ export default function sortLayers(map, layers, nextLayerId) {
         8. line  (other line layers that aren't primary layers)
         9. fill layers
   */
+
+  const moveLayers = filteredLayers => {
+    Object.keys(filteredLayers).forEach((key) => {
+      if (map.getLayer(filteredLayers[key].id)) {
+        map.moveLayer(filteredLayers[key].id);
+      }
+    });
+    if (
+      filteredLayers.find(d => d.id === nextLayerId) &&
+      map.getLayer(nextLayerId)
+    ) {
+      map.moveLayer(nextLayerId);
+    }
+  }
+
   if (Object.keys(layers).length) {
     const fill = Object.keys(layers)
       .map(d => layers[d])
       .filter(d => d.type === 'fill');
     if (fill.length) {
-      Object.keys(fill).forEach((key) => {
-        if (fill[key].id !== nextLayerId) {
-          if (map.getLayer(fill[key].id)) {
-            map.moveLayer(fill[key].id);
+      moveLayers(fill);
+    }
+
+    const heatMap = Object.keys(layers)
+      .map(d => layers[d])
+      .filter(d => d.type === 'heatmap');
+    if (heatMap.length) {
+      heatMap.sort((a,b) => b.priority - a.priority);
+      const checkPriority = Object.keys(heatMap).find(key => heatMap[key].priority)
+      if (checkPriority) {
+        Object.keys(heatMap).forEach((key) => {
```

Comment:
Interesting, is `priority` a part of the layer specification?

Reply:
@cKellyDesign ~ When we have many `heatmap` layers and a user may want to view them at once it is impossible to know in which order the user will turn them on so as to adjust the colors and opacity of the layers. Hence in such cases `priority` is used to always arrange the layers in a specific order.

Reply:
It feels like the order of the layers would normally be derived from the order the user turned on the individual layers. In the case of a grouped layer they should be rendered in the order they're set in the array.

Was this a specific request from the client?

Reply:
Not really it is something we thought working on UNOPs but now they have gone back to using grouped layers so let me do away with it.